motr/m0d.c

```diff
@@ -74,8 +74,17 @@ static bool regsignal = false;
  */
 static void cs_term_sig_handler(int signum)
 {
-	gotsignal = signum == SIGUSR1 ? M0_RESULT_STATUS_RESTART :
-					M0_RESULT_STATUS_STOP;
+	if (signum == SIGUSR1) {
+		gotsignal = M0_RESULT_STATUS_RESTART;
+	} else if (signum == SIGINT || signum == SIGQUIT || signum == SIGTERM) {
+		gotsignal = M0_RESULT_STATUS_STOP;
+	} else {
+		m0_console_printf("SIGNAL CAUGHT: %s", strsignal(signum));
+		do {
+			pause();
+		} while (!gotsignal);
```

Comment:
That's not right. A process can get SIGBUS for a variety of reasons unrelated to meta-data device failures. Even in the case of meta-data device failure, this can be a bad-block not a permanent failure. The proper way to handle a signal is to notify hare about failure (via m0_be_io_err_send()) and crash the process. Hare will restart it, if it crashes a few more times, hare will shutdown the process permanently.

The tricky point here is that m0_be_io_err_send() is not async safe and cannot be called from a signal handler.